src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -95,6 +99,28 @@ public static boolean shouldBeHearingsInWelshFlag() {
 
     public static List<HearingLocation> getHearingLocations(SscsCaseData caseData,
                                                             ReferenceDataServiceHolder referenceDataServiceHolder) {
+        List<HearingLocation> hearingLocations = getAllHearingLocations(caseData, referenceDataServiceHolder);
+
+        String nextHearingVenueName = caseData.getAdjournCaseNextHearingVenue();
+
+        if (!referenceDataServiceHolder.isAdjournmentFlagEnabled() && isNotEmpty(nextHearingVenueName)
+            && !HearingsChannelMapping.isPaperCase(caseData)) {
+            try {
+                String venueID = getVenueID(caseData, nextHearingVenueName);
+
+                return hearingLocations.stream()
+                    .filter(location -> location.getLocationId().equals(venueID))
+                    .collect(Collectors.toList());
+            } catch (InvalidMappingException e) {
+                log.error("Defaulting to all hearing locations {}", e.getMessage());
```

Comment:
I think it is desirable to include id of the case in the log message

Reply:
Included in the exception message

Reply:
ok